### PR TITLE
Challenge 3

### PR DESCRIPTION
**Pull request by machariamarigi**
**What this PR does**
Create A RESTFUL API using flask with the following features
- Create accounts for unregistered users
- Authenticate registered users with JWT tokens
- Users can CREATE, UPDATE and DELETE shopping lists
- Users can CREATE, UPDATE and DELETE items in shopping lists.
- Pagination and search queries for GET requests

**Testing**
Reviewers may test the application with `coverage run --source=api_v1 -m py.test && coverage report` from the root of the project for test results and coverage

**Running the Application**
Run `python run.py runserver` and visit `http://127.0.0.1:5000/api/v1` on the browser for the application documentation. Postman can also be used to see how the endpoints are working

### CODE REVIEW

File: api_v1/models.py [hunk 1]
```diff
@@ -70,22 +70,19 @@ def authenticate_password(self, password):
...
-                current_app.config.get('SECRET_KEY'),
-                algorithm='HS256'
-            )
-
-            return jwt_string
-        except Exception as excep:
-            return str(excep)
+        payload = {
+            'exp': datetime.utcnow() + timedelta(minutes=5),
+            'iat': datetime.utcnow(),
```
Comment (HawiCaesar) [inline api_v1/models.py]: Why this space?

**Review by HawiCaesar — COMMENTED** [review]
This commit deals with swaggerui configurations, how comes other files have been changed?

File: config.py [hunk 2]
```diff
@@ -8,6 +8,10 @@ class Config(object):
     SQLALCHEMY_DATABASE_URI = os.getenv('db_url')
     CSRF_ENABLED = True
     SECRET_KEY = os.getenv('secret')
+    SWAGGER_UI_DOC_EXPANSION = 'list'
```
Comment (HawiCaesar) [inline config.py]: This commit deals with swaggerui configurations, how comes other files have been changed?

File: tests/test_errors.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+"""Module for testing the running of the API"""
```
Comment (HawiCaesar) [inline tests/test_errors.py:1]: This is a test...What does it have to do with the commit? Was it affecting the test?

Comment (machariamarigi) [reply tests/test_errors.py:1]: Some of my changes were poorly committed. Thanks for pointing it out.

File: api_v1/models.py (hunk 1 above)
Comment (machariamarigi) [reply api_v1/models.py]: Please color

File: api_v1/endpoints.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    @token_required
+    def get(self, user_id):
+        """
+            Handle getting of all shoppinglists for an authorized user.
+            Resource Url --> /api/v1/shoppinglists
+        """
+        shoppinglists = Shoppinglist.query.filter_by(created_by=user_id)
+        results = []
+
+        for shoppinglist in shoppinglists:
```
Comment (HawiCaesar) [inline api_v1/endpoints.py]: I have seen that you have used backref and lazy=dyamic in your models. You may want to consider now bringing your shopping list items when you perform a GET shopping lists or GET shoppinglist. Something to think about

**Comment by HawiCaesar** [comment]
Good work in summary. Good boiler plate for your tests and having a 100% coverage. 
Refreshing your git commit knowledge will help avoid committing unnecessary files and making the commit look like its doing more than what was described. 
Considering working on how to solve the codacy issues

**Comment by Arthur236** [comment]
Nice test coverage 👌🏽

**Pull request merged**